packages/amplify-category-function/src/provider-utils/awscloudformation/index.ts

```diff
@@ -431,7 +433,14 @@ async function initTriggerEnvs(context, resourceParams, providerPlugin, envParam
     const currentTrigger = resourceParams.resourceName.replace(parentResourceParams.resourceName, '');
     if (currentTrigger && currentTrigger !== resourceParams.resourceName) {
       const currentEnvVariables = context.amplify.loadEnvResourceParameters(context, categoryName, resourceParams.resourceName);
-      const triggerPath = `${__dirname}/../../../../amplify-category-${resourceParams.parentStack}/provider-utils/${srvcMetaData.provider}/triggers/${currentTrigger}`;
+      const authPlugin = context.amplify.getCategoryPluginInfo(context, AmplifyCategories.AUTH, AmplifySupportedService.COGNITO);
```

Comment:
Is this method auth category specific? What about storage or other categories that enable triggers?